### ISSUE TITLE
[Hold for payment 2024-04-03][$500] Bank account - Different error is displayed on Validate you bank account page

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#37623]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.46-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4356707
**Email or phone of affected tester (no customers):** vdargentotest+web030124@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Pre-requisite: user must be logged in and have created a workspace
1. Go to Workspace > Bank account
2. Select Manual option
3. Go trough the add BA flow with the testing credentials until you reach the "Validate your bank account" page
4. Enter 3 random amounts (important step)
5. Click on "Validate" button
5. Verify the error is not the expected (Should be "The validate code you entered is incorrect, please try again")
6. Click on "fix the errors"
7. Verify it does not refocus to wrong field

## Expected Result:
The error displayed should be "The validate code you entered is incorrect, please try again"

## Actual Result:
The error displayed is "Please fix the errors in the form before continuing"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/59734ccf-e23b-41bb-b6a5-ec29dae3b4e4

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e383b39945f9cc66</li>
        <li>Upwork Job ID: 1763639802183884800</li>
        <li>Last Price Increase: 2024-03-01</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@luacmartins  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by luacmartins** [comment]
Looking

**Comment by luacmartins** [comment]
I don't think this is a blocker. We still show an error, just not a generic one instead of the specific error. Gonna demote and make it external

**Comment by FitseTLT** [comment]
BE

**Comment by strepanier03** [comment]
@sobitneupane - Looks like we have a proposal ready for your review. Thank you!

**Comment by luacmartins** [comment]
@sobitneupane have you had the chance to look at the proposal we have?

**Comment by strepanier03** [comment]
@sobitneupane - Friendly bump for proposal review please.

**Comment by sobitneupane** [comment]
Reviewing it now.

**Comment by sobitneupane** [comment]
@GandalfGwaihir Were you able to reproduce the issue? What  testing credentials did you use to reproduce the issue?

Can you please add more details on RCA of the problem? Which PR added typecheck? `typeof errorMessage === 'string'` Won't there be any regression on removing the typecheck?

**Comment by sobitneupane** [comment]
> Go trough the add BA flow with the testing credentials until you reach the "Validate your bank account" page

@lanitochka17 Can you please share the testing credentials that can be used in BA flow.

**Comment by strepanier03** [comment]
@sobitneupane - Little bump here, thanks!

**Comment by luacmartins** [comment]
waiting on @sobitneupane

**Comment by sobitneupane** [comment]
@strepanier03 @luacmartins I am still struggling to go through the steps to reproduce the issue. 

> Go trough the add BA flow with the testing credentials until you reach the "Validate your bank account" page

I have asked for help in slack https://expensify.slack.com/archives/C02NK2DQWUX/p1710325836952429

**Comment by sobitneupane** [comment]
> Go trough the add BA flow with the testing credentials until you reach the "Validate your bank account" page

@lanitochka17 Can you please share the testing credentials that can be used in BA flow to reproduce the issue.

**Comment by sobitneupane** [comment]
I was finally able to reproduce the issue. The issue is reproducible in Close account page as well.

**Comment by sobitneupane** [comment]
The following is the error abstracted from formstate.

`    const errorMessage = useMemo(() => (formState ? ErrorUtils.getLatestErrorMessage(formState) : undefined), [formState]);
`
```js
[
    "Looks like you have an outstanding balance! Please pay the balance owed before you try again.",
    {
        "isTranslated": true
    }
]
```

The type of errorMessage is object.

I have commented on https://github.com/Expensify/App/pull/37426/files#r1523196094 PR to find out why the type check was added and if we can safely remove it.

**Comment by situchan** [comment]
Please assign @kubabutkiewicz and me here since this came from https://github.com/Expensify/App/pull/37426

**Comment by sobitneupane** [comment]
@Krishna2323 [Your proposal](https://github.com/Expensify/App/issues/37623#issuecomment-1989182840) is not different from this [proposal](https://github.com/Expensify/App/issues/37623#issuecomment-1973803909)

**Comment by sobitneupane** [comment]
Discussion about the issue: https://github.com/Expensify/App/pull/37426#discussion_r1523196094

**Comment by sobitneupane** [comment]
After [discussion](https://github.com/Expensify/App/issues/37623#issuecomment-1973803909) in the issue, we are good to remove the type check. 

But the problem is we have two similar issues(https://github.com/Expensify/App/issues/38047) with the same root cause. I think we should merge these two issues.

cc: @luacmartins

**Comment by strepanier03** [comment]
@sobitneupane @GandalfGwaihir - Yes, #38047 is the same root cause and we can merge these two GHs leaving this primary but only if we're confident the rest of the resolution for this will be treated with urgency because #38047 is moving more quickly than this GH and we do care about moving forward quickly. 

Do you have a proposal you're ready to select and ask engineering to agree with? If not, we're going to let #38047 go forward as it's already at that stage. 

If you are at the stage where you're ready to get engineering to review and agree with your selected proposal please take action today. Tomorrow I'll be reviewing and having a final internal sense check before making a decision.

**Comment by sobitneupane** [comment]
Based on the [discussion](https://github.com/Expensify/App/issues/37623#issuecomment-1973803909) in the PR which introduced the type check, we are safe to remove the type check.
https://github.com/Expensify/App/blob/4f86857bd017177a84a7353918829cc969faebd6/src/components/Form/FormWrapper.tsx#L111

[Proposal](https://github.com/Expensify/App/issues/37623#issuecomment-1973803909) from @GandalfGwaihir looks good to me. There are other cases like Close Account Page which also has the same issue. Let's make sure to test those cases as well.

🎀 👀 🎀 C+ reviewed

**Comment by luacmartins** [comment]
I agree with @sobitneupane that @GandalfGwaihir's proposal has the correct RCA and solves the issue.

**Comment by strepanier03** [comment]
The other GH is on hold for this one.

**Comment by jayeshmangwani** [comment]
@GandalfGwaihir Issue is with Form component and it is a common component that used at many places,so instead of increasing bounty we should merge PR ASAP to avoid creating similar issues 😃 .

**Comment by luacmartins** [comment]
Added to the title. @strepanier03 could you please assist with payment?

**Comment by strepanier03** [comment]
Yup, handling tonight. The automation is being weird lately.

**Comment by strepanier03** [comment]
### Payment summary

- @GandalfGwaihir - $500 via Upwork - Waiting for you to accept the offer.
- @sobitneupane - $500 via Manual Requests in NewDot

@JmillsExpensify - Payment request incoming.

**Comment by JmillsExpensify** [comment]
$500 approved for @sobitneupane

**Comment by strepanier03** [comment]
@GandalfGwaihir - Sorry for the hold up, I have been OoO since last week. I'll work on the offer again now.

**Comment by strepanier03** [comment]
@GandalfGwaihir - I've just sent the offer again, can you take a look and I'll check again shortly to see if I can pay it out.

**Comment by strepanier03** [comment]
You're welcome @GandalfGwaihir - I just paid you out and closed the contract so everything is all set here. Closing now.
